### ISSUE TITLE
polynomial regression

### ISSUE THREAD

**Issue by fmeulen** [JuliaStats/StatsModels.jl#161]
While there is a complete example on implementing polynomial regression, it would be nice if this could be directly called like 
```
 fit(LinearModel, @formula(y ~ poly(x,5)), dataset)
```
(Ideally also with the possibility of using orthogonal polynomials.)

**Comment by kleinschmidt** [comment]
Agreed.  And splines, etc.  I think those features might belong in another package though (StatsModelsExtras.jl or something), possibly with the lead/lag terms, etc.

**Comment by HHammond** [comment]
I've been working on an implementation of regression splines (piecewise linear and restricted cubic splines with truncates power basis) that I'd be happy to contribute to a STatsModelsExtras module. I've been working on my own local repo.

**Comment by fmeulen** [comment]
Yes it would make perfectly sense to contribute that.
